plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3BlobStore.java

```diff
@@ -139,7 +91,7 @@ public String toString() {
     }
 
     public AmazonS3Reference clientReference() {
-        return service.client(repositoryMetadata);
+        return SocketAccess.doPrivileged(() -> service.client(repositoryMetadata));
```

Comment:
Hm ... why do we need `doPrivileged` here? The `S3Service::buildClient` has all necessary `SocketAccess.doPrivileged` calls (if not - we should fix that)?

Reply:
Yes you are right here, we shouldn't do a blanket privileged call here. Internally, we only need to make a privileged call for `RetryPolicy.builder()` since that performs a config file lookup. Removing this and adding privileged wrapper there.